Bookish/Controllers/CatalogController.cs

```diff
@@ -49,41 +59,161 @@ public IActionResult BookEntry(BookInputModel model)
     }
 
-    public void AddXCopies(int newCopies, LibraryContext context, Book book)
+    public void AddCopiesOfBook(int newCopies, LibraryContext context, Book book)
     {
+        List<Copy> newCopyList = new List<Copy>();
         for (int i = 0; i < newCopies; i++)
         {
             var copy = new Copy()
             {
                 BookID = book.BookID,
                 Comments = "This is a comment"
             };
-            context.Copies.Add(copy);
-            context.SaveChanges();
+            newCopyList.Add(copy);
         }
+        context.Copies.AddRange(newCopyList);
+        context.SaveChanges();
     }
-    
+    [HttpGet]
     public IActionResult BookList()
     {
         BookViewModel bvm = new BookViewModel();
         using (var context = new LibraryContext())
         {
-            bvm.EntryQuery = context.Books.Include(b => b.Author)
-                .Include(b => b.CopyList);
-            bvm.CatalogEntries = bvm.EntryQuery.ToList();
+            bvm.CatalogEntries = context.Books.Include(b => b.Author)
+                .Include(b => b.CopyList)
+                .ToList();
         }
-        return View(bvm);
+        return View("BookList",bvm);
     }
+    /*[HttpPost]
+    public IActionResult BookList(int bookId, string operation)
+    {
+        switch (operation)
+        {
+            case "delete":
+                throw new Exception("HELP DELETE");
+                return Delete(new BookViewModel());
+                break;
+            case "edit":
+                throw new Exception("HELP EDIT");
+                return Edit(new BookViewModel());
+                break;
+            default:
+                return View("Error");
+        }
+    }*/
+    [HttpGet]
+    public IActionResult Delete(int bookId)
+    {
 
+        Book? foundBook;
+        using (var context = new LibraryContext())
+        {
+            foundBook = context.Books.SingleOrDefault(a => a.BookID == bookId );
+            if (foundBook == null)
+            {
+                throw new Exception("NO BOOK!");
+            }
+            else
+            {
+                context.Remove(foundBook);
+                context.SaveChanges();
+            }
+        }
+
+        return BookList();
+    }
+    [HttpGet]
+    public IActionResult Edit(int bookId)
+    {
+        Book? foundBook;
+        using (var context = new LibraryContext())
+        {
+            foundBook = context.Books.SingleOrDefault(a => a.BookID == bookId);
+            foundBook.Author = context.Authors.SingleOrDefault(a => a.AuthorID == foundBook.AuthorID);
+            if (foundBook == null)
+            {
+                throw new Exception("NO BOOK!");
+            }
+        }
+        return View(foundBook);
+    }
+
+    [HttpPost]
+    public IActionResult Edit(Book b)
+    {
+        using (var context = new LibraryContext())
+        {
+            Book? foundBook = context.Books.SingleOrDefault(a => a.BookID == b.BookID);

```

Comment:
I'd use `b` rather than `a`, to indicate `Book` :)

Reply:
I've changed _b, b, and a to be more descriptive :')